avro/src/main/java/com/fasterxml/jackson/dataformat/avro/AvroMapper.java

```diff
@@ -126,4 +146,85 @@ public AvroSchema schemaFrom(File schemaFile) throws IOException
         return new AvroSchema(new Schema.Parser().setValidate(true)
                 .parse(schemaFile));
     }
+
+    public static class AvroSchemaTypeResolverBuilder extends StdTypeResolverBuilder {
+
+        @Override
+        public TypeSerializer buildTypeSerializer(SerializationConfig config, JavaType baseType, Collection<NamedType> subtypes) {
+            // No default typing for Serialization - It's already encoded in the Schema object
+            return null;
+        }
+
+        @Override
+        public TypeDeserializer buildTypeDeserializer(DeserializationConfig config, JavaType baseType, Collection<NamedType> subtypes) {
```

Comment:
That is essentially correct. Avro encodes polymorpic type information in the schema itself (as a `java-class` property). By setting the default typer, I check the schema to see what the actual type should be even when there is no available type information on the POJO for the `AnnotationIntrospector` to return.

If you peek in `InteropTestBase`, you will see that all I'm passing to Jackson as the target type is `Object.class`, a schema, and a byte array. All additional type information comes from the schema, which is how the Apache implementation encodes it.